### PR TITLE
Replace abandoned toast library with react-aria component

### PR DESCRIPTION
**Pull request by eikhr**
# Description

We were getting deprecation warnings because [`react-notification`](https://www.npmjs.com/package/react-notification) hasn't been updated in 5 years. I replaced it with a component based on [`react-aria` toasts](https://react-spectrum.adobe.com/react-aria/useToast.html).

# Result

Minor visual changes, but I've tried to keep it similar. 

Screenshot 2024-10-12 at 16 03 52
Screenshot 2024-10-12 at 16 03 17

# Testing

- [x] I have thoroughly tested my changes.

---

Resolves ABA-1121

### CODE REVIEW

**Review by ivarnakken — APPROVED** [review]

**Comment by ivarnakken** [comment]
Looks much much better! 😍 Awesome!!

File: app/components/Toast/Toast.module.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    left: 0;
+    right: 0;
+    gap: 0;
+  }
+}
+
+.toast {
+  display: flex;
+  align-items: center;
+  background-color: var(--color-gray-9);
```
[the remark this answers is not in the document]
Comment (eikhr) [reply app/components/Toast/Toast.module.css:24]: We could use `var(--lego-font-color)` for the background, but I thought it might be kind of bad practice to use the font color for non-font stuff. `var(--lego-card-color)` is opposite of what we want i believe (toast is dark in light mode and light in dark mode)

[the remark this answers is not in the document]
Comment (ivarnakken) [reply app/components/Toast/Toast.module.css:24]: I believe we use the font color for tooltips

File: app/components/Toast/Toast.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      onAnimationEnd={() => {
+        // Remove the toast when the exiting animation completes.
+        if (props.toast.animation === 'exiting') {
+          state.remove(props.toast.key);
+        }
+      }}
+    >
+      <Button {...closeButtonProps} className={styles.closeButton}>
+        <Icon iconNode={<XIcon />} size={18} />
+      </Button>
```
[the remark this answers is not in the document]
Comment (eikhr) [reply app/components/Toast/Toast.tsx]: Yep, it uses `onPress`...

[the remark this answers is not in the document]
Comment (ivarnakken) [reply app/components/Toast/Toast.tsx]: I suppose the icon should support both, but I guess we can fix that later. I like when icon buttons act as buttons

[the remark this answers is not in the document]
Comment (eikhr) [reply app/components/Toast/Toast.tsx]: https://github.com/webkom/lego-webapp/pull/5054

[the remark this answers is not in the document]
Comment (ivarnakken) [reply app/components/Toast/Toast.tsx]: Removing `onClick` is probably fine 🤷🏼‍♂️ Not sure if I get why `onPress` is the new cool thing

[the remark this answers is not in the document]
Comment (eikhr) [reply app/components/Toast/Toast.tsx]: They have a lot of reasoning behind it in `react-aria-components` https://react-spectrum.adobe.com/react-aria/interactions.html
TLDR is that it is an abstraction that works better for all types of press events (clicks, taps, keyboard presses, etc.)

**Review by jonasdeluna — APPROVED** [review]
Nice but is there not too little spacing between subsequent toasts?
